### PR TITLE
feat: override checkstyle version through dedicated configuration

### PR DESCRIPTION
**Pull request by mamadzz**
Issue: https://github.com/ekino/gradle-quality-plugin/issues/10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,3 +99,12 @@ plugins {
 ### Requirements
 
 Gradle 5.3.1 and JDK 8 are required.
+
+
+You can override checkstyle version using a dedicated configuration (default is 8.24):
+```kotlin
+checkstyle {
+    toolVersion = "8.20"
+    configFile = file("${project.rootDir}/config/checkstyle.xml")
```
Comment (clemstoquart) [inline README.md:110]: Hi,
Thanks for the PR.
Can you please add a note about the fact that the default `checkstyle.xml` is compatible with `8.24` or later.

File: src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt [hunk 2]
```diff
@@ -97,8 +92,9 @@ class QualityPlugin: Plugin<Project> {
     }
   }
 
-  private fun getTextResource(project: Project, file: String): TextResource {
-    return project.resources.text.fromString(QualityPlugin::class.java.getClassLoader().getResource(file).readText())
+  private fun getCheckstyleConfig(): File {
+    return QualityPlugin::class.java.classLoader?.getResource("checkstyle.xml")?.path
+            ?.let { File(it) }!!
```
Comment (clemstoquart) [inline src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt]: I think we should throw an exception with an accurate error message if the config file can't be found.

**Review by clemstoquart — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mamadzz) [reply README.md:110]: Hi, 
Yes i think it's a good idea.
Thank you.

File: src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt (hunk 2 above)
Comment (mamadzz) [reply src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt]: Yes it's better with an exception with an explicit message.
Thank you.

**Review by clemstoquart — APPROVED** [review]

**Pull request merged**